server/controllers/userAuth.js

```diff
@@ -511,42 +508,13 @@ const deleteAccount = async (req, resp) => {
 };
 
 const prepareItems = async (req, resp, next) => {
-  const { _id: userId } = req.user;
-  const data = { requests: [], todos: [] };
+  const { user } = req;
 
   try {
-    const ownerLists = await List.find({
-      ownerIds: { $in: [userId] },
-      isArchived: false,
-      'items.0': { $exists: true }
-    })
-      .lean()
-      .populate('cohortId', 'name')
-      .populate('items.authorId', 'displayName')
-      .exec();
-
-    const viewerLists = await List.find({
-      viewersIds: { $in: [userId] },
-      isArchived: false,
-      'items.0': { $exists: true }
-    })
-      .lean()
-      .populate('cohortId', 'name')
-      .populate('items.authorId', 'displayName')
-      .exec();
-
-    if (ownerLists) {
-      data.todos = prepareTodosItems(ownerLists);
-    }
-
-    if (viewerLists) {
-      data.requests = prepareRequestedItems(viewerLists)(userId);
-    }
+    const reportData = await getItemsForReport(List, user);
```

Comment:
unnecessary List argument, you can simplify this by passing just `getItemsForReport(user) `
and within ` helper.js`  file you can also have access to` List.model ` by requiring at the top of the file

Reply:
Should helpers file be connected with models? I know that this "helper" is closely associated with List but thought that helpers should rely on the params passed in the place of use? What do you think?

Reply:
okay you're right. I agree with point 1.